spec/solver_spec.rb

```diff
@@ -0,0 +1,31 @@
+require_relative '../solver'
+
+describe Solver do
+  context 'When testing the Solver class' do
+    it 'Should create a new solver when we call the class method new' do
+      new_solver = Solver.new
+      expect(new_solver.class.to_s).to eq 'Solver'
+    end
+    it 'Should return the factorial for a given number' do
+      new_solver = Solver.new
+      expect(new_solver.factorial(4)).to eq 24
+    end
+  end
+  context 'When testing the Solver reverse method' do
+    it 'Should return a reverse string' do
+      new_solver = Solver.new
+      expect(new_solver.reverse('hello')).to eq 'olleh'
+      expect(new_solver.reverse('juanito')).to eq 'otinauj'
+      expect(new_solver.reverse('google')).to eq 'elgoog'
+    end
+  end
+  context 'When testing the fizzbuzz method' do
+    it 'Should return a string' do
+      new_solver = Solver.new
+      expect(new_solver.fizzbuzz(15)).to eq 'fizzbuzz'
+      expect(new_solver.fizzbuzz(9)).to eq 'fizz'
+      expect(new_solver.fizzbuzz(10)).to eq 'buzz'
+      expect(new_solver.fizzbuzz(4)).to eq 'Its not divisble by 3 nor 5'
+    end
+  end
+end
```

Comment:
- [ ] Make sure to test each method in its own `describe` block for readability and clarity, you tested for the `Solver` class and the `#factorial` method in the same block. For example, the scaffold for your tests should look like 👇🏼 
```ruby
require_relative '../solver'

describe Solver do
    # you can test for the Solver class here
    # you can use contexts if necessary

    describe "#factorial"
        # your assertions here
    end

    describe "#reverse"
        # your assertions here
    end

    describe "#fizzbuzz"
        # your assertions here
    end
end
```
- [ ] Remember to test for both positive and negative scenarios, e.g for the `#factorial` method, a negative value should raise an exception.

- [ ] [OPTIONAL] For the `#fizzbuzz` method, you can divide the tests into separate `context` blocks for readability and clarity, e.g, you can add `context`s when the number is both divisible by `5` and `3`, when it's only divisible by `5`, when it's only divisible by `3`, and when it's not divisible by neither.

Reply:
Thank you cyonii, now we fix the code for a better readability and clarity 😄